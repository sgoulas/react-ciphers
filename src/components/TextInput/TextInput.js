import React from "react";
import { withStyles, makeStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";

const CssTextField = withStyles({
  root: {
    width: 450,
    "& label.Mui-focused": {
      color: "#264653",
    },
    "& .MuiInput-underline:after": {
      borderBottomColor: "#264653",
    },
    "& .MuiOutlinedInput-root": {
      "& fieldset": {
        borderColor: "#264653",
      },
      "&:hover fieldset": {
        borderColor: "#E76F51",
      },
      "&.Mui-focused fieldset": {
        borderColor: "#E76F51",
      },
    },
  },
})(TextField);

const useStyles = makeStyles((theme) => ({
  root: {
    display: "flex",
    flexWrap: "wrap",
  },
  margin: {
    margin: theme.spacing(1),
  },
}));

const TextInput = (props) => {
  const { text, setText } = props;
  const classes = useStyles();

  const handleChange = (event) => {
    const { value } = event.target;
    setText(value);
  };

  return (
    <CssTextField
      className={classes.margin}
      label="Start typing"
      id="plain-text-input"
      value={text}
      onChange={handleChange}
    />
  );
};

export default TextInput;
