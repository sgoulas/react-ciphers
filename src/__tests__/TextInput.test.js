import React from "react";
import renderer from "react-test-renderer";
import TextInput from "../components/TextInput/TextInput";

describe("TextInput suite", () => {
  it("renders correctly", () => {
    const textInput = renderer
      .create(<TextInput text="" setText={() => {}} />)
      .toJSON();

    expect(textInput).toMatchSnapshot();
  });
});
