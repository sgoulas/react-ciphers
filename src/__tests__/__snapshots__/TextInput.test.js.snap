// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput suite renders correctly 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-3 makeStyles-margin-2"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
    data-shrink={false}
    htmlFor="plain-text-input"
    id="plain-text-input-label"
  >
    Start typing
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      id="plain-text-input"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value=""
    />
  </div>
</div>
`;
